src/views/Viewer.vue

```diff
@@ -707,6 +714,12 @@ export default {
 		onClose() {
 			this.Viewer.onClose()
 		},
+
+		async onDelete() {
+			const url = this.root + this.currentFile.filename
+			await axios.delete(url)
+			this.close()
```

Comment:
I'm wondering isn't it more intuitive to show a previous file rather than closing the viewer?

Reply:
Make sense, good idea!
Any ideas how I can change the file to previous after deletion?

Reply:
In line 627 there is a method called `previous()`. Could you please check if this is what we are looking for? I see this code for the first time :)

Reply:
https://github.com/nextcloud/viewer/blob/master/src/views/Viewer.vue#L41
If you do this, you  also have to remove the deleted file from the files list.

Reply:
Implemented it.
It's working without changing the files list 👍

Reply:
It's working without changing the files list +1

If you do this, you will have an error because pressing next will try to open the deletd file

Reply:
OK, now the file should be removed from fileList